density/config/flask_config.py

```diff
@@ -11,31 +10,69 @@
 from sys import exit
 
 
-try:
-    # flask settings
-    HOST = environ['HOST']
-    PORT = environ['PORT']
-    SECRET_KEY = environ['SECRET_KEY']
-    DEBUG = True if environ['DEBUG'] == 'TRUE' else False
-
-    # postgres settings
-    PG_USER = environ['PG_USER']
-    PG_PASSWORD = environ['PG_PASSWORD']
-    PG_DB = environ['PG_DB']
-    PG_HOST = environ['PG_HOST']
-    PG_PORT = environ['PG_PORT']
-
-    # oauth settings
-    GOOGLE_CLIENT_ID = environ['GOOGLE_CLIENT_ID']
-
-    if not DEBUG:
-        MAIL_SERVER = 'smtp.gmail.com'
-        MAIL_PORT = 465
-        MAIL_USE_SSL = True
-        MAIL_USE_TLS = False
-        MAIL_DEFAULT_SENDER = 'densitylogger@gmail.com'
-        MAIL_USERNAME = 'densitylogger@gmail.com'
-        MAIL_PASSWORD = environ['MAIL_PASSWORD']
+# dictionary the flask app configures itself from
+config = {
+    'HOST': '0.0.0.0',
+    'PORT': None,
+    'SECRET_KEY': None,
+    'PG_USER': None,
+    'PG_PASSWORD': None,
+    'PG_DB': None,
+    'PG_HOST': None,
+    'PG_PORT': None,
+    'GOOGLE_CLIENT_ID': None,
+    'DEBUG': True if environ.get('DEBUG') == 'TRUE' else False
+}
+
+# consul_configurations contains equivalent keys that will be used to extract
+# configuration values from Consul.
+consul_configurations = [  # consul key --> config key
+    ('flask_port', 'PORT'),
+    ('secret_key', 'SECRET_KEY'),
+    ('postgres_user', 'PG_USER'),
+    ('postgres_password', 'PG_PASSWORD'),
+    ('postgres_database', 'PG_DB'),
+    ('postgres_host', 'PG_HOST'),
+    ('postgres_port', 'PG_PORT'),
+    ('google_client_id', 'GOOGLE_CLIENT_ID'),
+]
+
+if config.get('DEBUG'):
+
+    try:  # use local settings
+        for k, v in config.iteritems():
+            if not v:
+                config[k] = environ[k]
+
+    except KeyError as e:
+        """ Throw an error if a setting is missing """
+        print "ERR MSG: {}".format(e.message)
+        print ("Some of your settings aren't in the environment."
+               "You probably need to run:"
+               "\n\n\tsource config/<your settings file>")
+        exit(1)
+
+else:  # prod w/ consul
+    from consul import Consul
+    kv = Consul().kv  # initalize client to KV store
+
+    for k, v in consul_configurations:
+        _, tmp = kv.get("density/{}".format(k))  # density is the root key
+        val = tmp.get('Value')
+        config[v] = val
+        if not val:
+            raise Exception("no val found in Consul for density/{}".format(k))
+
+    # mail settings
+    MAIL_SERVER = 'smtp.gmail.com'
+    MAIL_PORT = 465
+    MAIL_USE_SSL = True
+    MAIL_USE_TLS = False
+    MAIL_DEFAULT_SENDER = 'densitylogger@gmail.com'
+    MAIL_USERNAME = 'densitylogger@gmail.com'
+    _, MAIL_PASSWORD = kv.get('density/mail_password')
```

Comment:
When this key doesn't exist, you'll get a `TypeError`, because it will try to unpack `None`.  Here's an example:

```
>>> _, a = {}.get('k')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'NoneType' object is not iterable
```